dags/inference/maxtext_inference.py

```diff
@@ -16,204 +16,490 @@
-          "sleep_time": 120,
-          "tpu_version_cores": [(TpuVersion.V5E, 8), (TpuVersion.V5P, 8)],
-          "checkpoint": "gs://inference-benchmarks/models/llama2-7b/2024-04-25-14-01/param-only-decode-ckpt-maxtext/checkpoints/0/items",
-          "model_mode": "base",
-          "maxtext_logs": "gs://inference-benchmarks/models/llama2-7b/2024-04-25-14-01/",
-          "scan_layers": "false",
-          "dataset": "openorca",
-          "weight_dtype": "bfloat16",
+  test_name_prefix = "max-js" if not USER_PREFIX else f"{USER_PREFIX}-max-js"
+
+  test_templates = {
+      # LLAMA2_7B
+      LLAMA2_7B: {
+          "maxtext_branch": maxtext_branch,
+          "jetstream_branch": jetstream_branch,
+          "sleep_time": 360,
+          "time_out_in_min": 120,
+          "tpu_version_cores": [(TpuVersion.V5E, 8)],#, (TpuVersion.V5E, 4)],
+          "model_name": LLAMA2_7B,
           "tokenizer": "tokenizer.llama2",
-          "per_device_batch_sizes": [1, 2, 4, 8, 11, 12],
-          # (ici_fsdp_parallelism, ici_autoregressive_parallelism, ici_tensor_parallelism)
-          "ici_parallelisms": [(1, -1, 1), (1, 1, -1)],
-          "request_rate": 5,
-          "num_prompts": 1000,
+          "weight_dtype": "bfloat16",
+          "scan_layers": "false",
           "max_prefill_predict_length": 1024,
           "max_target_length": 2048,
+          "reshape_q": True,
+          # (ici_fsdp_parallelism, ici_autoregressive_parallelism, ici_tensor_parallelism)
+          "ici_parallelisms": [(1, 1, -1)],
+          "dataset": "openorca",
+          "num_prompts": 1000,
           "max_output_length": 1024,
+          "warmup_mode": "full",
       },
-      "llama2-13b": {
-          "sleep_time": 120,
-          "tpu_version_cores": [(TpuVersion.V5E, 8), (TpuVersion.V5P, 8)],
-          "checkpoint": "gs://inference-benchmarks/models/llama2-13b/2024-04-25-14-01/param-only-decode-ckpt-maxtext/checkpoints/0/items",
-          "model_mode": "base",
-          "maxtext_logs": "gs://inference-benchmarks/models/llama2-13b/2024-04-25-14-01/",
-          "scan_layers": "false",
-          "dataset": "openorca",
-          "weight_dtype": "bfloat16",
+      f"{LLAMA2_7B}-{W_BF16_KV_BF16}-dot-product": {
+          "attention": "dot_product",
+          "request_rate": [0.0],
+          "axis_order": [
+              "0123-2013-2013",
+              "0213-0213-0213",
+              "0213-0213-0132",
+          ],
+      },
+      f"{LLAMA2_7B}-{W_INT8_KV_INT8}-dot-product": {
+          "attention": "dot_product",
+          "request_rate": [0.0],
+          "axis_order": [
+              "0213-0213-0213",
+              "0213-0231-0213",
+          ],
+      },
+      # LLAMA2_13B
+      LLAMA2_13B: {
+          "maxtext_branch": maxtext_branch,
+          "jetstream_branch": jetstream_branch,
+          "sleep_time": 360,
+          "time_out_in_min": 120,
+          "tpu_version_cores": [(TpuVersion.V5E, 8)],#, (TpuVersion.V5E, 4)],
```

Comment:
Running v5e-8 for now. Need additional logic for combining smaller chip and smaller batch size. 

@morgandu Should we just merge this only for v5e-8? And add in v5e-4 in a subsequent PR?

Reply:
Removed v5e-4 for now.